File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    let row = matrix.length;
    let col = matrix[0].length;
    let rows = [];
    let cols = [];
    for(let i=0;i<row;i++){
        for(let j=0;j<col;j++){
            if(matrix[i][j] === 0){
                rows[i] = true;
                cols[j] = true;
            }
        }
    }
    for(let i=0;i<row;i++){
        if(rows[i]){
            for(let j=0;j<col;j++){
                matrix[i][j] = 0;
            }
        }
    }
    for(let i=0;i<row;i++){
        for(let j=0;j<col;j++){
            if(cols[j]){
                matrix[i][j] = 0;
            }
        }
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int row = matrix.length;
        int col = matrix[0].length;
        boolean[] rows = new boolean[row];
        boolean[] cols = new boolean[col];
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(matrix[i][j] == 0){
                    rows[i] = true;
                    cols[j] = true;
                }
            }
        }
        // rows zero
        for(int i=0;i<row;i++){
            if(rows[i])
                for(int j=0;j<col;j++){
                    matrix[i][j] = 0;
                }
        }
        // cols zero
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(cols[j])
                    matrix[i][j] = 0;
            }
        }
    }
}